analysis/topEFT/run_sow.py

```diff
@@ -0,0 +1,184 @@
+parser.add_argument('--treename'       , default='Events', help = 'Name of the tree inside the files')
+parser.add_argument('--xrd'            , default='', help = 'The XRootD redirector to use when reading directly from json files')
+parser.add_argument('--wc-list'        , action='extend', nargs='+', help = 'Specify a list of Wilson coefficients to use in filling histograms.')
+
+args = parser.parse_args()
+inputFiles = args.inputFiles.replace(' ','').split(',')  # Remove whitespace and split by commas
+chunksize  = args.chunksize
+nchunks    = args.nchunks if args.nchunks else None
+outname    = args.outname
+outpath    = args.outpath
+treename   = args.treename
+xrd        = args.xrd
+max_files  = args.max_files
+wc_lst     = args.wc_list if args.wc_list is not None else []
+
+samples_to_process = {}
+for fn in inputFiles:
+    if fn.endswith('.json'):
+        sample = os.path.basename(f).replace('.json','')
+        jsn = load_sample_json_file(fn)
+        samples_to_process = update_cfg(jsn,
+            name=sample,
+            cfg=samples_to_process,
+            max_files=max_files,
+            redirector=xrd
+        )
+    elif fn.endswith('.cfg'):
+        samples_to_process = read_cfg_file(fn,cfg=samples_to_process,max_files=max_files)
+    else:
+        raise RuntimeError(f"Unknown input file: {fn}")
+
+flist = {}
+#for sample_name,jsn in samples_to_process['jsons'].items():
+for sample_name,jsn in samples_to_process.items():
+    xrd_src = jsn['redirector']
+    flist[sample_name] = [f"{xrd_src}{fn}" for fn in jsn['files']]
+
+    # Basically only try to build up the WC list if we haven't manually specified a list to use
+    if len(wc_lst) == 0:
+        for wc in jsn['WCnames']:
+            if wc not in wc_lst:
+                wc_lst.append(wc)
+
+    jsn_txt = json.dumps(jsn,indent=2,sort_keys=True)
+
+    # Strips off the leading and closing curly brackets, along with any associated newlines
+    jsn_txt = jsn_txt[1:-1].strip("\n",)
+
+    s = ""
+    s += f">> {sample_name}\n"
+    s += f"{jsn_txt}\n"
+    print(s)
+
+if wc_lst:
+    if len(wc_lst) == 1:
+        wc_print = ", ".join(wc_lst)
+    else:
+        wc_print = ", ".join(wc_lst[:-1]) + f", and {wc_lst[-1]}"
+    print(f"Wilson Coefficients: {wc_print}.")
+else:
+    print("No Wilson coefficients specified")
+
+processor_instance = sow_processor.AnalysisProcessor(samples_to_process,wc_lst)
+
+executor_args = {
+    'master_name': '{}-workqueue-coffea'.format(os.environ['USER']),
+
+    'xrootdtimeout': 900,
+
+    # find a port to run work queue in this range:
+    'port': [9123,9130],
+
+    'debug_log': 'debug.log',
+    'transactions_log': 'tr.log',
+    'stats_log': 'stats.log',
+
+    'environment_file': topeftenv.get_environment(),
+    'extra_input_files': ["sow_processor.py"],
+
+    'schema': NanoAODSchema,
+    'skipbadfiles': False,
+
+    'wrapper': "python_package_run_with_retries",
```

Comment:
This should not be needed, I think.

Reply:
this is a good point. I'll swap the wrapper back to the original.